src/events/EventsSection.tsx

```diff
@@ -122,60 +230,269 @@ export const EventsSection = () => {
+                    </DropdownToggle>
+                  }
+                  isOpen={searchDropdownOpen}
+                >
+                  <Form
+                    isHorizontal
+                    className="keycloak__user_events_search__form"
+                    data-testid="searchForm"
+                  >
+                    <FormGroup
+                      label={t("userId")}
+                      fieldId="kc-userId"
+                      className="keycloak__user_events_search__form_label"
+                    >
+                      <TextInput
+                        ref={register()}
+                        type="text"
+                        id="kc-userId"
+                        name="userId"
+                        data-testid="userId-searchField"
+                        onChange={(_val, event) => {
+                          const { value } = event.currentTarget;
+                          setValue("userId", value);
+                          setIsDirty(true);
+                        }}
+                        defaultValue={
+                          formValues?.userId ? formValues?.userId : ""
+                        }
+                      />
+                    </FormGroup>
+                    <FormGroup
+                      label={t("eventType")}
+                      fieldId="kc-eventType"
+                      className="keycloak__user_events_search__form_label"
+                    >
+                      <Select
+                        id="kc-eventType"
+                        name="eventType"
+                        data-testid="event-type-searchField"
+                        chipGroupProps={{
+                          numChips: 1,
+                          expandedText: "Hide",
+                          collapsedText: "Show ${remaining}",
+                        }}
+                        variant={SelectVariant.typeaheadMulti}
+                        typeAheadAriaLabel="Select"
+                        onToggle={onSelectToggle}
+                        selections={selectedEvents}
+                        onSelect={(_, value) => {
+                          const option = value.toString();
+                          register("eventTypes");
+                          if (selectedEvents?.includes(option)) {
+                            setSelectedEvents(
+                              selectedEvents.filter((item) => item !== option)
+                            );
+                            setValue(
+                              "eventTypes",
+                              selectedEvents.filter((item) => item !== option)
+                            );
+                            setIsDirty(true);
+                          } else {
+                            setSelectedEvents([...selectedEvents, option]);
+                            setValue("eventTypes", [...selectedEvents, option]);
+                            setIsDirty(true);
+                          }
```

Comment:
There is some duplicated logic here, I think it can be shortened.
```suggestion
                          const selected = selectedEvents.includes(option)
                            ? selectedEvents.filter((item) => item !== option)
                            : [...selectedEvents, option];

                          setSelectedEvents(selected);
                          setValue("eventTypes", selected);
                          setIsDirty(true);
```

Reply:
true, awesome, thanks Jon!